kube/src/config/mod.rs

```diff
@@ -91,11 +91,12 @@ impl Config {
     /// and relies on you having the service account's token mounted,
     /// as well as having given the service account rbac access to do what you need.
     pub fn from_cluster_env() -> Result<Self> {
-        let cluster_url = incluster_config::kube_server().ok_or(ConfigError::MissingInClusterVariables {
-            hostenv: incluster_config::SERVICE_HOSTENV,
-            portenv: incluster_config::SERVICE_PORTENV,
-        })?;
-        let cluster_url = cluster_url.parse::<http::Uri>()?;
+        let cluster_url = incluster_config::kube_dns();
+        //let cluster_url = incluster_config::kube_server().ok_or(ConfigError::MissingInClusterVariables {
+        //    hostenv: incluster_config::SERVICE_HOSTENV,
+        //    portenv: incluster_config::SERVICE_PORTENV,
+        //})?;
+        //let cluster_url = cluster_url.parse::<http::Uri>()?;
```

Comment:
If this is a sound method, then a question is how do we roll this out.

Evar to opt in to new dns method? I am a bit scared to jump to the dns method that client-go doesn't even use on a hard-swap.

Reply:
rolling it out for rustls only (because that's broken anyway).